src/addrman.h

```diff
@@ -166,6 +166,9 @@ class CAddrInfo : public CAddress
 //! the maximum number of tried addr collisions to store
 #define ADDRMAN_SET_TRIED_COLLISION_SIZE 10
 
+//! the maximum time we'll spend trying to resolve a tried table collision, in seconds
+static const int64_t ADDRMAN_TEST_WINDOW = 20*60; // 20 minutes
```

Comment:
Unless I'm missing something this should be set to a value higher than 20 minutes.

1. A feeler connection is made according to a Poisson distribution with an average of 2 minutes. 50% of the time the interval will take more than 2 minutes. Additionally this interval does not include the connection time, one if each address in the buffer times out.
2. The collision set buffer can hold at least 10 entries.

Thus at 20 minutes it is very possible that a value might be added to the collision set buffer and not be tried for more than 20 minutes and then evicted. I would set this to at least 40 minutes. This won't eliminate the chance an untried entry is evicted, but it will reduce it.

Sadly I don't have time to do the math, but if someone else has a few minutes, could someone compute the probability that a address would sit in the collision set for 40 minutes and not be tried. Assume that each prior attempt times out.

Reply:
I haven't looked into this in detail, but wanted to point out that the chance that the time between 2 Poisson-distributed events has only a 1/e = 36.8% chance of being larger than the average interval.

Reply:
Oh, is the question what the probability is that the sum of 10 exponentially-distributed variables (independent, each with average 2 minutes) exceeds 20 minutes?

https://www.wolframalpha.com/input/?i=P(X%3E20)+where+X~GammaDistribution(10,1%2F2) says 0.00000039%.

With 40 minutes it would be 0.00000000000000000000075%.

Reply:
Oops, used the wrong arguments on Wolfram in the previous comment.

Is the question what the probability is that the sum of 10 exponentially-distributed variables (independent and each with an average of 2 minutes) exceeds 20 minutes?

https://www.wolframalpha.com/input/?i=P(X%3E20)+where+X~GammaDistribution(10,2) says 45.8%.

Exceeding 40 minutes has a chance of 0.5%.

Reply:
How likely is it that we add things to `m_tried_collisions` more often than once every 2 minutes or so?

Reply:
>How likely is it that we add things to `m_tried_collisions` more often than once every 2 minutes or so?

Not sure how to compute that probability but I can contrive a scenario where it happens:

1. Assume a victim nodes tried table is full of good nodes, and the victim nodes new table is full of attacker nodes. 

2. Victim makes 8 outgoing connections and gets unlucky and chooses all 8 from new. This means the victim tries to move these 8 IPs in new to tried causing 8 collisions (remember we assumed tried is full). All 8 of these connections are being made to attacker nodes (since all new addresses are attack controlled).

3. Attacker then breaks each of these 8 connections so the victim must reconnect. The victim chooses at least 2 nodes from new and causes two new collisions when it tries to add them to tried. Now there are 10 addresses in `m_tried_collisions`.


You could probably trigger that behavior by filling the new table with your IP addresses, letting the node connect to you, add you to tried (if tried was filled causing a collision) and then

Reply:
I've updated to 40 minutes, thanks.